job_crafter.py

```diff
@@ -102,75 +111,96 @@ def make_jobs(self):
+                    defconfigs = test.get('defconfigs', self.board['defconfigs'])
+
             for defconfig in defconfigs:
-                if not self.options['kernel']:
-                    # No custom kernel, go fetch artifacts on remote locations
-                    data = (ArtifactsFinder("http://lava.free-electrons.com/downloads/builds/",
-                            **self.options).crawl(self.board, defconfig) or
-                            ArtifactsFinder("https://storage.kernelci.org/",
-                            **self.options).crawl(self.board, defconfig))
+                logging.info("  Configuring defconfig: %s" % defconfig)
+                data = None
+
+                for url in ("http://lava.free-electrons.com/downloads/builds/",
+                            "https://storage.kernelci.org/"):
+                    finder = ArtifactsFinder(self.cfg, url)
+                    try:
+                        data = finder.crawl(self.board, defconfig)
+                    except IOError:
+                        logging.debug("Didn't find the artifacts on server %s" % url)
+
+                if data is None:
+                    logging.error("No artifacts available, bailing out")
+                    raise IOError
 
                 job_name = "%s--%s--%s--%s" % (
                         self.board['device_type'],
-                        self.options["tree"],
+                        self.cfg['tree'],
                         defconfig,
                         test['name']
                         )
 
                 self.override('kernel', data.get('kernel'))
+                logging.info("    Kernel path: %s" % self.job['kernel'])
+
                 self.override('device_tree', data.get('dtb'))
+                logging.info("    Device tree path: %s" % self.job['device_tree'])
+
                 self.override('modules', data.get('modules'))
+                logging.info("    Modules archive path: %s" % self.job['modules'])
 
                 self.get_template_from_file(os.path.join(TEMPLATE_FOLDER,
                     test.get('template', DEFAULT_TEMPLATE)))
 
                 self.job["tests"] = test['name']
-                print("tests: Overridden")
-                self.job["job_name"] = self.options['job_name'] or job_name
-                print("job name: %s" % self.job["job_name"])
+
+                if 'job_name' in self.cfg:
+                    self.job["job_name"] = self.cfg['job_name']
+                else:
+                    self.job["job_name"] = job_name
+                logging.debug("    Job name: %s" % self.job['modules'])
 
                 # Complete job creation
-                if self.options["no_send"]:
+                if self.cfg['no_send']:
                     self.save_job_to_file()
                 else:
                     self.send_to_lava()
 
-    def override(self, key, value=None):
-        if value:
-            print("%s: Overriding with file:" % key, value)
```

Comment:
I think those lines should stay in `logging.debug`. They are quite useful for development.